frontend/src/App.jsx

```diff
@@ -37,22 +41,61 @@ function App() {
     if (searchTerm) {
       try {
         const formattedSearchTerm = searchTerm.replace(/[^a-zA-Z ]/g, "");
-        const receivedImages = await getSearchedImages(imageType, formattedSearchTerm);
+        const receivedImages = await getSearchedImages(imageType, formattedSearchTerm, limit, offset);
 
         setImages(receivedImages);
+        updateOffsetValue();
         setIsLoading(false);
       } catch (error) {
         console.log(error);
       }
       return;
     }
 
-    const receivedImages = await getImages(imageType);
+    const receivedImages = await getImages(imageType, limit, offset);
 
     setImages(receivedImages);
+    updateOffsetValue();
     setIsLoading(false);
   }
 
+  const loadMoreImages = async () => {
+    if (inputValue) {
+      try {
+        const formattedSearchTerm = inputValue.replace(/[^a-zA-Z ]/g, "");
+        const receivedImages = await getSearchedImages(imageType, formattedSearchTerm, limit, offset);
+
+        setImages([...images, ...receivedImages]);
+        updateOffsetValue();
+        setIsLoading(false);
+      } catch (error) {
+        console.log(error);
+      }
+      return;
+    }
+
+    const receivedImages = await getImages(imageType, limit, offset);
+
+    setImages([...images, ...receivedImages]);
+    updateOffsetValue();
+  }
+
+  const setInitialOffsetValue = (value) => {
+    if (value === 'gifs') {
+      setOffset(0);
+    } else {
+      setOffset(1);
+    }
+  }
+
+  const updateOffsetValue = () => {
+    if (imageType === 'gifs') {
```

Comment:
Can we use ternary operator here? I think it is more readable that way. What do you think ?